lib/index.js

```diff
@@ -94,6 +94,11 @@ Server.prototype.checkRequest = function(req, fn) {
 Server.prototype.serveClient = function(v){
   if (!arguments.length) return this._serveClient;
   this._serveClient = v;
+
+  if(v && !clientSource) {
+      clientSource = read(require.resolve('socket.io-client/socket.io.js'), 'utf-8');
```

Comment:
I agree with @palavrov, side effects are much easier avoided using this method, and for what it's worth I can't see it being an improvement to do it otherwise.

Please add a space between the `if` and the following `(` though, and also indent with two spaces not four, just to be picky 😉